src/keyDerivation_test.c

```diff
@@ -0,0 +1,141 @@
+#include "keyDerivation.h"
+#include "test_utils.h"
+#include "hex_utils.h"
+#include "stream.h"
+#include <string.h>
+#include "errors.h"
+
+#define ARRAYLEN(array) (sizeof(array) / sizeof(array[0]))
+
+void testPrivateKeyDerivation()
+{
+#define TESTCASE(path_, pathLength_, expected_) \
+	{ \
+		uint8_t chainCode[32]; \
+		cx_ecfp_256_extended_private_key_t privateKey; \
+		derivePrivateKey(path_, pathLength_, chainCode, &privateKey); \
+		stream_t s; \
+		stream_init(&s); \
+		stream_appendFromHexString(&s, expected_); \
+		const uint8_t* expectedBuffer = stream_head(&s); \
+		EXPECT_EQ_BYTES(expectedBuffer, privateKey.d, 64); \
+	}
+
+	{
+		uint32_t path1[] = { HARDENED_BIP32 + 44, HARDENED_BIP32 + 1815, HARDENED_BIP32 + 1};
+		char* expected1 = "5878bdf06af259f1419324680c4d4ce05b4343895b697144d874749925e69d41"
+		                  "648e7e50061809207f81dac36b73cd8fab2325d49d26bc65ebbb696ea59c45b8";
+
+		TESTCASE(path1, ARRAYLEN(path1), expected1);
```

Comment:
Note: TESTCASE macro can do arraylen internally (although this would be different if it wasn't a macro but a function call...)